Extend form docs with example form errors

tordans opened the pull request:
Forms should show a error message whenever there is an error. This can be done nicely in bootstrap with the `.text-help` class. But until know people hat to figure that out by themselves.

This commit extends the example html with dummy error messages for the warning and danger cases. It also shows a case where a text-help paragraph is combined with an text-muted paragraph, wich requires an additional `m-b-0` at the moment (see https://github.com/twbs/bootstrap/issues/18703).
In addition to that I extended the introduction to specify the `.text-help` class as the right way to show error messages in general. The `.text-help` is not explained in the form-docs yet, but it should.

This is what the new docs look like:
<img width="840" alt="bildschirmfoto 2015-12-27 um 14 51 49" src="https://cloud.githubusercontent.com/assets/111561/12010485/6c1f6900-aca9-11e5-847a-c4144b3db946.png">

jamesmosier committed:
Modal docs: Added JS event handler for relatedTarget demo

Fixes #17584
Closes #17607 by merging it

tordans committed:
Document the possibility of extending the $spacers map

It's very handy to add another spacer size (e.g. `4`) which can easily be done by extending the map.
Example:
```
$spacers: (
  0: (
    x:   0,
    y:   0
  ),
  1: (
    x:   $spacer-x,
    y:   $spacer-y
  ),
  2: (
    x:   ($spacer-x * 1.5),
    y:   ($spacer-y * 1.5)
  ),
  3: (
    x:   ($spacer-x * 3),
    y:   ($spacer-y * 3)
  ),
  4: (
    x:   ($spacer-x * 5),
    y:   ($spacer-y * 5)
  )
);
```

Closes #18706
[skip sauce]

cvrebert commented on docs/components/forms.md:
```diff
@@ -551,6 +551,8 @@ Block help text—for below inputs or for longer lines of help text—can be eas
 
 Bootstrap includes validation styles for danger, warning, and success states on form controls. To use, add `.has-warning`, `.has-danger`, or `.has-success` to the parent element. Any `.form-control-label`, `.form-control`, and `.text-help` within that element will receive the validation styles.
 
+This makes `.text-help` the right choise to display error messages in your forms.
```
Spelling: "choice", not "choise" [sic]

cvrebert commented:
@mdo The existence of `.text-help` contradicts the statement in http://v4-alpha.getbootstrap.com/components/forms/#help-text that "No official help text classes exist in Bootstrap 4". Could we get a ruling on which is supposed to be correct here?

Commit:
Document .pos-f-t utility class; fixes #18494

Add documentation to components/utilities to publicly expose the
.pos-f-t utility class to users.

Closes #18799
[skip sauce]

jwkratz committed:
Docs: Add example of .dropdown-menu-right

Fixes #18493
Closes #18836
[skip sauce]

XhmikosR committed:
Update grunt-html to v6.0.0.

Also remove the workaround for the old validator version.

[skip sauce]

XhmikosR committed:
Fix validation errors.

[skip sauce]

TiGR committed:
Fix docs for new naming of .btn-outline

This one was overlooked at edit

See #17719 and #19199

cvrebert committed:
Merge pull request #19204 from TiGR/patch-1

Fix docs for new naming of .btn-outline
[skip sauce]

cvrebert committed:
Extract new $modal-dialog-margin variable

[skip sauce]
[skip validator]

cvrebert committed:
Merge pull request #19205 from twbs/modal-dialog-margin

Extract new $modal-dialog-margin variable

cvrebert committed:
Extract $modal-dialog-sm-up-margin-y variable

[skip sauce]
[skip validator]

cvrebert committed:
Merge pull request #19206 from twbs/modal-dialog-sm-up-margin-y

Extract $modal-dialog-sm-up-margin-y variable

cvrebert committed:
Extract $modal-{header,footer}-border-width variables

[skip sauce]
[skip validator]

cvrebert committed:
Merge pull request #19207 from twbs/modal-header-footer-width-vars

Extract $modal-{header,footer}-border-width variables

cvrebert committed:
Extract $modal-content-border-width variable

[skip sauce]
[skip validator]

cvrebert committed:
Merge pull request #19208 from twbs/modal-content-border-width

Extract $modal-content-border-width variable

cvrebert committed:
Add $mark-padding variable

[skip sauce]
[skip validator]

cvrebert committed:
Add $small-font-size variable

[skip sauce]
[skip validator]

cvrebert committed:
Introduce $mark-bg variable so that <mark>'s background-color can be customized independently of $state-warning-bg

[skip sauce]
[skip validator]

cvrebert committed:
Extract $label-padding-{x,y} variables

[skip sauce]
[skip validator]

cvrebert committed:
Extract $label-pill-padding-x variable

[skip sauce]
[skip validator]

cvrebert committed:
Merge pull request #19212 from twbs/label-padding-x-y

Extract $label-padding-{x,y} variables

cvrebert committed:
Add $label-font-size variable in the name of not hardcoding values

[skip sauce]
[skip validator]

cvrebert committed:
Add $label-pill-border-radius variable

[skip sauce]
[skip validator]

cvrebert committed:
Merge pull request #19213 from twbs/label-pill-padding-x

Extract $label-pill-padding-x variable

cvrebert committed:
Extract $dropdown-min-width variable

[skip sauce]
[skip validator]

cvrebert committed:
Merge pull request #19216 from twbs/dropdown-min-width

Extract $dropdown-min-width variable

cvrebert committed:
Change comment to clarify that $grid-breakpoints are minimum widths for each breakpoint

(as opposed to maximums)

[ci skip]

cvrebert committed:
CONTRIBUTING: Document restriction regarding jQuery event alias methods

Refs #19192
[ci skip]

cvrebert committed:
Merge pull request #19219 from twbs/contrib-jq-event-aliases

CONTRIBUTING: Document restriction regarding jQuery event alias methods

cvrebert committed:
js/tests/index.html: Replace jQuery event alias methods with functions that throw explanatory errors, instead of with undefined

Refs https://github.com/twbs/bootstrap/pull/19219#issuecomment-184653699

[skip validator]

cvrebert committed:
Merge pull request #19220 from twbs/jq-event-alias-throw-err

js/tests/index.html: Replace jQuery event alias methods with functions that throw errors

patrickhlauke committed:
Focus (visually hidden) input radio/checkbox

This allows for keyboard navigation (e.g. arrow keys left/right to
trigger different radio buttons) following an initial mouse click
interaction

patrickhlauke committed:
Merge pull request #19222 from patrickhlauke/kbd-navigation-fix-checkbox-radio-toggle-port1

Focus (visually hidden) input radio/checkbox

kandros committed:
Fix typos in plural tag names

patrickhlauke committed:
Visual/manual test for checkbox/radio button kbd behavior

cvrebert committed:
Merge pull request #19224 from kandros/v4-dev

Fix typos in documentation - in plural tag names in reboot.md
[skip sauce]

patrickhlauke committed:
Merge pull request #19232 from patrickhlauke/tests-visual-buttons-patch1

Visual/manual test for checkbox/radio button kbd behavior

mdo committed:
grunt

mdo committed:
fixes #19148 as follow up to #19099: ensure local mixin variable is used for  in make-grid-columns mixin

mdo committed:
Fixes #19221: Add .w-100 as width: 100% utility class

mdo committed:
Merge pull request #19218 from twbs/breakpoint-vars-are-minimums

Clarify that $grid-breakpoints are minimum widths for each breakpoint

mdo committed:
Merge pull request #19215 from twbs/label-pill-border-radius

Add $label-pill-border-radius variable

mdo committed:
Merge pull request #19214 from twbs/label-font-size

Add $label-font-size variable in the name of not hardcoding values

mdo committed:
Merge pull request #19211 from twbs/mark-bg-var

Add $mark-bg so <mark>'s background-color can be customized independently of $state-warning-bg

mdo committed:
Merge pull request #19210 from twbs/small-font-size

Add $small-font-size variable

mdo committed:
grunt

mdo committed:
Merge branch 'v4-dev' into mark-padding

mdo committed:
Merge pull request #19116 from twbs/docs-17264

Docs: Add captions above progress bars

mdo committed:
Merge branch 'fix-19094' of https://github.com/ameya-pandilwar/bootstrap into ameya-pandilwar-fix-19094

mdo committed:
grunt

mdo committed:
update migration docs

mdo committed:
Fixes #19142 (follow-up to #18771): Restore cursor: pointer to custom radio/checkbox components

mdo committed:
grunt

cvrebert committed:
Extract $carousel-control-sm-up-size variable

[skip sauce]
[skip validator]

cvrebert committed:
Merge pull request #19235 from twbs/carousel-control-sm-up-width

Extract $carousel-control-sm-up-width variable

cvrebert committed:
Extract $carousel-indicator{-active}-size variables

[skip sauce]
[skip validator]

cvrebert committed:
Extract $dropdown-padding-y variable

[skip sauce]
[skip validator]

cvrebert committed:
Extract $dropdown-margin-top variable

[skip sauce]
[skip validator]

cvrebert committed:
Extract $dropdown-item-padding-x variable

[skip sauce]
[skip validator]

cvrebert committed:
Merge pull request #19240 from twbs/dropdown-item-padding-x

Extract $dropdown-item-padding-x variable

cvrebert committed:
Extract $carousel-indicators-width variable

[skip sauce]
[skip validator]

cvrebert committed:
Extract $carousel-caption{-sm-up}-width variables

[skip sauce]
[skip validator]

cvrebert committed:
Extract $list-group-item-heading-margin-bottom variable

[skip sauce]
[skip validator]

cvrebert committed:
Add $figure-caption-font-size variable

[skip sauce]
[skip validator]

helloworld123122344134 committed:
_tags.scss: Updated comments to reflect name change

Erratum from #19157
Closes #19245

cvrebert committed:
Extract $code-font-size variable

[skip sauce]
[skip validator]

cvrebert committed:
Merge pull request #19244 from twbs/figure-caption-font-size

Add $figure-caption-font-size variable

cvrebert committed:
Merge pull request #19246 from twbs/code-font-size

Extract $code-font-size variable

cvrebert committed:
Extract $code-padding-{x,y} variables

[skip sauce]
[skip validator]

cvrebert committed:
.navbar-divider width: 1px=>$border-width

[skip sauce]
[skip validator]

cvrebert committed:
Merge pull request #19247 from twbs/code-padding-vars

Extract $code-padding-{x,y} variables

cvrebert committed:
Add $list-group-item-padding-{x,y} variables

[skip sauce]
[skip validator]

cvrebert committed:
Add $card-columns-sm-up-column-gap variable

[skip sauce]
[skip validator]

cvrebert committed:
Merge pull request #19249 from twbs/card-columns-sm-up-column-gap

Add $card-columns-sm-up-column-gap variable

cvrebert committed:
Merge pull request #19250 from twbs/list-group-item-padding

Add $list-group-item-padding-{x,y} variables

cvrebert committed:
Extract $navbar-brand-padding-y variable

[skip sauce]
[skip validator]

cvrebert committed:
Merge pull request #19251 from twbs/navbar-brand-padding-y

Extract $navbar-brand-padding-y variable

patrickhlauke committed:
- Conveying color meaning to AT as markdown include
- Add/replace callout for color and AT with include

patrickhlauke committed:
Replace "label" with "tag" in navigation

patrickhlauke committed:
Merge pull request #19256 from patrickhlauke/label-tag-nav-fixup1

Replace "label" with "tag" in navigation

cvrebert committed:
Add $gutter parameter to make-col mixin; fixes #19147

[skip sauce]
[skip validator]

cvrebert committed:
Merge pull request #19257 from twbs/fix-19147

Add $gutter parameter to make-col mixin

patrickhlauke committed:
Replace jQuery with vanilla JS

patrickhlauke committed:
Merge pull request #19259 from patrickhlauke/keyboard-navigation-focus-patch1

Replace jQuery with vanilla JS

patrickhlauke committed:
Merge pull request #19255 from patrickhlauke/color-accessibility-callout-include

Docs: add/unify advice on contextual colors and accessibility

mdo committed:
Merge pull request #19237 from twbs/carousel-indicator-size

Extract $carousel-indicator{-active}-size variables

mdo committed:
Merge pull request #19239 from twbs/dropdown-margin-top

Extract $dropdown-margin-top variable

mdo committed:
Merge pull request #19242 from twbs/carousel-caption-width

Extract $carousel-caption-width variable

cvrebert committed:
Merge pull request #19243 from twbs/list-group-item-heading-margin-bottom

Extract $list-group-item-heading-margin-bottom variable

cvrebert committed:
Merge pull request #19238 from twbs/dropdown-padding-y

Extract $dropdown-padding-y variable

cvrebert committed:
Merge pull request #19241 from twbs/carousel-indicators-width

Extract $carousel-indicators-width variable

cvrebert committed:
Docs: Remove <meta name="keywords"> from <head>

Ports #18872 to v4
[skip sauce]

cvrebert committed:
Merge pull request #18987 from xcatliu/v4-dev-fix-search-json-prefix

Add prefix to footer.html search.json
[skip sauce]

XhmikosR committed:
Travis CI: switch to the Trusty image.

XhmikosR committed:
Merge pull request #19029 from twbs/v4-dev-xmr-grunt-html

Update grunt-html

XhmikosR committed:
Bump scss_lint to v0.45.0.

[skip sauce]

XhmikosR committed:
Update devDependencies.

XhmikosR committed:
Run grunt.

[ci skip]

patrickhlauke committed:
Appropriate use of fieldset/div, legend, callout for hidden/no label alternatives

patrickhlauke committed:
Add form-control-legend

XhmikosR committed:
Update scss_lint to v0.46.0.

[skip sauce]

XhmikosR committed:
Update Jekyll to v3.1.2.

[skip sauce]

XhmikosR committed:
Bump grunt-contrib-concat.

[skip sauce]

cvrebert committed:
Use Windows 10 for Win Chrome+Firefox tests on Sauce Labs

[skip validator]

cvrebert committed:
Merge pull request #19291 from twbs/win10

Use Windows 10 for Win Chrome+Firefox tests on Sauce Labs

cvrebert committed:
Use OS X El Capitan for Sauce tests

[skip validator]

cvrebert committed:
Use iOS 9.2 in Sauce tests

[skip validator]

cvrebert committed:
Merge pull request #19293 from twbs/el-capitan

Use OS X El Capitan for Sauce tests

cvrebert committed:
Merge pull request #19294 from twbs/ios-9.2

Use iOS 9.2 in Sauce tests

prose100 committed:
Correct grammar

cvrebert committed:
Merge pull request #19312 from prose100/docs

docs/components/alerts.md: Correct grammar
[ci skip]

patrickhlauke committed:
Merge pull request #19277 from patrickhlauke/v4-dev-forms-fixes#2

Forms fixes: fieldset/legend, new .form-control-legend, accessibility advice

cvrebert committed:
Wall of Browser Bugs: Update Chromium bug tracker URLs

RIP code.google.com
[skip sauce]

cvrebert committed:
Merge pull request #19321 from twbs/bugs.chromium.org

Wall of Browser Bugs: Update Chromium bug tracker URLs

Commit:
Include all grunt requires files in npm package

XhmikosR committed:
Bump scss_lint to v0.47.0.

[skip sauce]

XhmikosR committed:
Travis CI: Specify node.js 5.6 since 5.7.0 has regressions.

This should be reverted when the next 5.x version is out.

[skip sauce]

XhmikosR committed:
Merge pull request #19328 from satsukitv/v4-dev

Include all grunt required files in npm package

XhmikosR committed:
Update anchor-js to v3.1.0.

[ci skip]

XhmikosR committed:
Update clipboard.js to v1.5.9.

[ci skip]

XhmikosR committed:
Update jekyll-search to v1.1.1.

[ci skip]

XhmikosR committed:
Update tether to v1.2.0.

[skip sauce]
[skip validator]

XhmikosR committed:
Run `grunt`.

[ci skip]

imsky committed:
update theme parameters to fix #19333

cvrebert committed:
Merge pull request #19333 from twbs/v4-dev-update-docs-assets

Update docs assets

helloworld123122344134 committed:
Removed unused markdown-it dependency

cvrebert committed:
Merge pull request #19337 from ameya-pandilwar/fix-19332

Removed unused markdown-it dependency

XhmikosR committed:
Update holder.js to v2.9.3.

[ci skip]

XhmikosR committed:
Update npm-shrinkwrap.json.

[skip sauce]

XhmikosR committed:
Merge pull request #19331 from twbs/v4-dev-fix-travis

Travis: use node.js 5.6

XhmikosR committed:
Run grunt.

[ci skip]

cvrebert committed:
JS visual tests: Remove usages of defunct .page-header class

[skip sauce]

cvrebert committed:
Merge pull request #19340 from twbs/js-tests-page-header

JS visual tests: Remove usages of defunct .page-header class

cvrebert committed:
Example pages: Remove usages of defunct .page-header class

/fyi @mdo

Note: The selector in dashboard.css that used .page-header was superfluous
    because the margin that it zeroed was already zero even without that declaration.

[skip sauce]

cvrebert committed:
Merge pull request #19341 from twbs/examples-page-header

Example pages: Remove usages of defunct .page-header class

MaartenvE committed:
Added colon before enumeration

XhmikosR committed:
Update tether CDN.

[ci skip]

XhmikosR committed:
Revert "Travis CI: Specify node.js 5.6 since 5.7.0 has regressions."

This reverts commit d0059ac2ed5335e1d68b161639f7c8f619378cc7.

Node.js 5.7.1 is out with the issue fixed.

[skip sauce]

XhmikosR committed:
Update grunt-jscs.

[skip sauce]

XhmikosR committed:
Bump scss_lint.

[skip validator]
[skip sauce]

cvrebert committed:
Merge pull request #19348 from delftswa2016/grammar

Added colon before enumeration
[skip sauce]

cvrebert committed:
Add Wall of Browser Bugs entry for #19390

https://bugzilla.mozilla.org/show_bug.cgi?id=1253468
Closes #19390
[skip sauce]

cvrebert committed:
Merge pull request #19393 from twbs/mozbug-1253468

Add Wall of Browser Bugs entry for #19390

cvrebert committed:
Remove Wall of Browser Bugs entry for #19390

Refs https://bugzilla.mozilla.org/show_bug.cgi?id=1217012
Refs #19393
[ci skip]

cvrebert committed:
Merge pull request #19395 from twbs/moz-1253468-fixed-ff45

Remove Wall of Browser Bugs entry for #19390

XhmikosR committed:
Remove the now unused btoa dependency.

[skip sauce]

patrickhlauke committed:
Neutralize link styles for placeholder links/named anchors

patrickhlauke committed:
Merge pull request #19411 from patrickhlauke/v4-links-not-anchors

Neutralize link styles for placeholder links/named anchors

cvrebert committed:
_reboot.scss: Capitalization & periods in comment

For consistency with the rest of the file
[ci skip]

florisverburg committed:
Introduction for using table-reflow.

patrickhlauke committed:
Merge pull request #19408 from delftswa2016/v4-dev

Introduction for using table-reflow

cvrebert committed:
Docs: Add link to Chrome bug regarding `body { overflow: hidden; }`

https://bugs.chromium.org/p/chromium/issues/detail?id=175502
[skip sauce]

cvrebert committed:
Merge pull request #19423 from twbs/crbug-175502

Docs: Add link to Chrome bug regarding `body { overflow: hidden; }`

cvrebert committed:
Docs: Link directly to printing-related browser bugs

* https://bugs.chromium.org/p/chromium/issues/detail?id=273306
* https://bugs.webkit.org/show_bug.cgi?id=138192
[skip sauce]

cvrebert committed:
Merge pull request #19424 from twbs/browser-print-bugs

Docs: Link directly to printing-related browser bugs

cvrebert committed:
Remove iOS virtual keyboard bug section from docs

The modal backdrop positioning bug related to iOS' virtual keyboard doesn't seem to reproduce in iOS 8.0+.
(Possibly as a side-effect of https://bugs.webkit.org/show_bug.cgi?id=153224 )
Refs #9023
[skip sauce]

cvrebert committed:
Add jQuery and Tether to npm dependencies; fixes #17325

[skip sauce]

XhmikosR committed:
Bump devDependencies.

Also switch to `^`.

XhmikosR committed:
Run `grunt`.

[ci skip]

cvrebert committed:
Merge pull request #19404 from twbs/v4-dev-xmr-remove-btoa

Remove the now unused btoa dependency.

cvrebert committed:
Travis CI: Remove now-unused $GEMDIR env var

[skip sauce]

cvrebert committed:
Merge pull request #19431 from twbs/bye-gemdir

Travis CI: Remove now-unused $GEMDIR env var

patrickhlauke committed:
Add native title tooltip IE11/Edge bug to wall of browser bugs

patrickhlauke committed:
Merge pull request #19434 from patrickhlauke/v4-dev-ie-edge-tooltip-browser-bug

Add native title tooltip IE11/Edge bug to wall of browser bugs

XhmikosR committed:
Examples: Update tether to v1.2.0.

[ci skip]

XhmikosR committed:
Remove sass gem.

Jekyll 3.x has it as a dependency already.

cvrebert committed:
Update Wall of Browser Bugs entry for #15990

http://bugzil.la/1139853 was resolved as a duplicate of http://bugzil.la/924068
Refs #15990
[ci skip]

cvrebert committed:
Wall of Browser Bugs: Stop mentioning IE11

Sounds like IE11 isn't getting non-critical fixes anymore, so there's no point in mentioning it (just like there's no point in mentioning IE10).
Help us, Edge-Wan Kenobi! You're our only hope.
[ci skip]

cvrebert committed:
Merge pull request #19452 from twbs/ie11-wall-bye

Wall of Browser Bugs: Stop mentioning IE11

wolfy1339 committed:
Port #19469 to v4

Make hound fail on violations

cvrebert committed:
CONTRIBUTING.md: Prohibit "+1" comments in favor of reactions

Refs https://github.com/blog/2119-add-reactions-to-pull-requests-issues-and-comments
Refs https://github.com/dear-github/dear-github/issues/141#issuecomment-195106918
[ci skip]

neilhem committed:
remove empty sass if directives

helloworld123122344134 committed:
Enabled ESLint and JSCS in Hound

Fixes #19114
Closes #19482, #19299

cvrebert committed:
Merge pull request #19470 from twbs/plus-1-stfu

CONTRIBUTING.md: Prohibit "+1" comments in favor of reactions
[ci skip]

davidlormor committed:
Document $enable-grid-classes variable

Closes #19532
[skip sauce]

cvrebert committed:
Merge pull request #19426 from twbs/ios-virt-keyboard

Remove iOS virtual keyboard bug section from docs

cvrebert committed:
Mention https://webkit.org/b/153856 in the docs

Refs #14839, #19088
[skip sauce]

cvrebert committed:
Merge pull request #19537 from twbs/wkbug-153856

Mention https://webkit.org/b/153856 in the docs

cvrebert committed:
Merge pull request #19427 from twbs/npm-deps

Add jQuery and Tether to npm dependencies

XhmikosR committed:
Remove the now unused glob.

cvrebert committed:
Add Wall of Browser Bugs entry for #17338

https://crbug.com/595210

Closes #17338
[skip sauce]

cvrebert committed:
Merge pull request #19551 from twbs/crbug-595210

Add Wall of Browser Bugs entry for #17338
[ci skip]

cvrebert committed:
Use "latest" version specifier (when applicable) in Sauce tests

Refs http://sauceio.com/index.php/2016/03/new-browser-version-shortcuts/
[skip validator]

cvrebert committed:
Merge pull request #19552 from twbs/sauce-latest

Use "latest" version specifier (when applicable) in Sauce tests

cvrebert committed:
Add Android to Sauce testing

[skip validator]

cvrebert committed:
Merge pull request #19549 from twbs/v4-dev-xmr-rm-glob

Remove the now unused glob devDependency

cvrebert committed:
Merge pull request #19553 from twbs/sauce-android

Add Android to Sauce testing
[skip validator]

XhmikosR committed:
Bump jekyll-redirect-from.

[skip sauce]

craigmichaelmartin committed:
Docs: Call out removal of @screen-* variables in v4

Closes #19561
Closes #19567
[skip sauce]

XhmikosR committed:
Update jQuery to v2.2.2.

XhmikosR committed:
Update dependencies.

[ci skip]

mdo committed:
Merge pull request #19473 from neilhem/remove-empty-if

remove empty sass if directives

mdo committed:
Make active navbar link weight 500 again

mdo committed:
lighter active

mdo committed:
don't shrink nav font-size

mdo committed:
grunt

cvrebert committed:
Merge pull request #19407 from twbs/v4-dev-xmr-jquery-221

Update jQuery to v2.2.2.

cvrebert committed:
Use vanilla npm shrinkwrap instead of uber/npm-shrinkwrap; fixes #18559

cvrebert committed:
update shrinkwrap

cvrebert committed:
Merge pull request #19603 from twbs/bump-shrinkwrap

Update shrinkwrap
[skip sauce]
[skip validator]

cvrebert committed:
update shrinkwrap

cvrebert committed:
Merge pull request #19604 from twbs/fix-18559

Use vanilla npm shrinkwrap instead of uber/npm-shrinkwrap

XhmikosR committed:
Update grunt-html.

[skip sauce]

cvrebert committed:
Add Wall of Browser Bugs entry for #19626

Refs https://bugzilla.mozilla.org/show_bug.cgi?id=1259972
Closes #19626
[ci skip]

cvrebert committed:
Merge pull request #19628 from twbs/ff-bug-1259972

Add Wall of Browser Bugs entry for #19626

XhmikosR committed:
Nuke grunt-line-remover.

Use concat's process function instead.

KayLeung committed:
Docs: Correct arg `offsets` to `offset` in tooltips & popovers

[skip sauce]

Commit:
Finish killing .center-block; fixes #19443

Replace the leftover instances with .m-x-auto (and .d-block when necessary)
Closes #19665
Refs #19102
[skip sauce]

Commit:
Add `display` utility classes (.d-{inline,block,inline-block}); fixes #19343

Refs #19665
[skip sauce]

cvrebert committed:
CONTRIBUTING.md: Update MS Edge bug reporting instructions

[ci skip]

cvrebert committed:
Reflect MS Edge's migration of their issue tracker

Refs https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/
[skip sauce]

cvrebert committed:
Reflect MS Edge's migration of their issue tracker

Refs https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/
[ci skip]

cvrebert committed:
Wall of Browser Bugs: Update some more URLs of MS Edge bugs

[skip sauce]

cvrebert committed:
Update Wall of Browser Bugs entry for #14211

Refs https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/5381673/
[skip sauce]

lionelerard committed:
Update offset class name in comment in _grid-framework.scss

[skip sauce]
[skip validator]

cvrebert committed:
Add docs for .w-100 utility class

[skip sauce]

cvrebert committed:
Merge pull request #19234 from twbs/v4-w100

Add `.w-100` as width: 100% utility class

cvrebert committed:
Update lingering implicit prose reference to `.center-block`

Closes #19339
Refs #19102, #19443, #19665
[skip sauce]

cvrebert committed:
https://webkit.org/b/143527 has been fixed

Refs #16180, #16262
[skip sauce]

mdo committed:
Fixes #19690: Mention classes better for .list-inline and .list-inline-item

patrickhlauke committed:
Add aria-label and callout about labelling pagination components

patrickhlauke committed:
Merge pull request #19705 from patrickhlauke/v4-pagination-a11y-label

Add aria-label and callout about labelling pagination components

cvrebert committed:
Update 300ms click delay docs to reflect iOS 9.3 improvements

[skip sauce]

cvrebert committed:
https://crbug.com/273306 has been fixed

Refs #12078
[skip sauce]

cvrebert committed:
Update remaining Edge issue URLs in Wall of Browser Bugs

* https://connect.microsoft.com/IE/feedback/details/2445370 => https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/6793560/
* https://connect.microsoft.com/IE/feedback/details/2256049 => https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/7165383/
[skip sauce]

cvrebert committed:
docs/_plugins/bugify.rb: Remove IE; all relevant issues moved to Edge

[skip sauce]

cvrebert committed:
Merge pull request #19712 from twbs/crbug-273306

https://crbug.com/273306 has been fixed
[skip sauce]

cvrebert committed:
Merge pull request #19713 from twbs/edge-urls

Update remaining Edge issue URLs in Wall of Browser Bugs

patrickhlauke committed:
Merge pull request #19710 from twbs/ios-9.3-fast-tap

Update 300ms click delay docs to reflect iOS 9.3 improvements

cvrebert committed:
Update shrinkwrap

cvrebert committed:
grunt dist

[ci skip]

patrickhlauke committed:
Suppress IE/Edge additional white/blue colors for focused <select>

cvrebert committed:
Add http://bugzil.la/1264125 to Wall of Browser Bugs

Due to the lack of this feature in browsers,
we currently have to implement an annoying setTimeout-based workaround,
which adds an extra layer of complication and probably slightly impacts performance:
https://github.com/twbs/bootstrap/blob/e391fcb953cc959ddd88fb676b68f49eef85ef06/js/src/util.js#L63-L77

[ci skip]

cvrebert committed:
Merge pull request #19727 from twbs/bugzilla-1264125

Add http://bugzil.la/1264125 to Wall of Browser Bugs
[ci skip]

cvrebert committed:
https://webkit.org/b/128489 has been fixed!

Refs #12536
[ci skip]

cvrebert committed:
Merge pull request #19737 from twbs/wkbug-128489

Remove https://webkit.org/b/128489 from Wall of Browser Bugs

cvrebert committed:
Port #19741 to v4

[ci skip]

cvrebert committed:
Merge pull request #19248 from twbs/navbar-divider-border-width

.navbar-divider width: 1px=>$border-width

Rplus committed:
remove unnecessary `/` in reference url

patrickhlauke committed:
Merge pull request #19756 from Rplus/v4-dev

remove `/` in reference url (otherwise 404s)

mdo committed:
grunt

mdo committed:
Merge branch 'v4-dev' of https://github.com/twbs/bootstrap into v4-dev

mdo committed:
grunt

cvrebert committed:
Add Wall of Browser Bugs entries for https://webkit.org/b/156684 & https://webkit.org/b/156687

These bugs are factors in our decisions regarding which unit to use in our media queries.
Refs #17403
[skip sauce]

cvrebert committed:
Merge pull request #19765 from twbs/mq-wkbugs

Add Wall of Browser Bugs entries for WebKit media query bugs
[skip sauce]

cvrebert committed:
Modal: Remove IE8-specific window.innerWidth workaround

[skip validator]

cvrebert committed:
Merge pull request #19775 from twbs/modal-ie8-innerWidth

Modal: Remove IE8-specific window.innerWidth workaround

patrickhlauke committed:
Merge pull request #19714 from patrickhlauke/v4-ie-edge-select-normalisation

Suppress IE/Edge additional white/blue colors for focused <select>

tordans committed:
Merge remote-tracking branch 'origin/v4-dev' into v4-dev

twbs-rorschach commented:
Hi @tordans!

Thanks for wanting to contribute to Bootstrap by sending this pull request!
Unfortunately, your pull request seems to have some problems:
* [Changes must be made to the original Sass or Less source code file(s), not just the compiled CSS file(s).](https://github.com/twbs/rorschach/blob/master/docs/css.md)

You'll need to **fix these mistakes** and revise your pull request before we can proceed further.
Once you've fixed these problems, you can either ask the maintainers to re-open this pull request, or you can create a new pull request.
Thanks!

(*Please note that this is a [fully automated](https://github.com/twbs/rorschach) comment.*)
